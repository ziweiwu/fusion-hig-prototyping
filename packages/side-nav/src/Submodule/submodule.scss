@import "~@hig/styles/mixins/colors.scss";
@import "~@hig/styles/mixins/spacing.scss";
@import "~@hig/styles/mixins/typography.scss";

.hig__side-nav__submodule {
  @include typography-base;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-bottom: 10px;
  padding-right: spacing(m);
  text-decoration: none;
  cursor: pointer;
  user-select: none;

  &:first-child {
    margin-top: spacing(xs);
  }
  &:last-child {
    margin-bottom: spacing(m);
  }

  &:focus,
  &:active {
    outline: none;
  }
}

// Light theme
.hig--light-theme {
  &.hig__side-nav__submodule {
    color: color(hig-cool-gray-50);
    svg * {
      stroke: color(hig-cool-gray-50);
    }

    &:hover,
    &:focus,
    &:active,
    &--active,
    &--active:focus {
      color: color(hig-blue-50);
      svg * {
        stroke: color(hig-blue-50);
      }
    }
  }
}

// Dark blue theme
.hig--dark-blue-theme {
  &.hig__side-nav__submodule {
    color: transparentize($color: color(hig-white), $amount: 0.25);
    svg * {
      stroke: transparentize($color: color(hig-white), $amount: 0.25);
    }

    &:hover,
    &--active,
    &--active:focus {
      color: color(hig-blue-40);
      svg * {
        stroke: color(hig-blue-40);
      }
    }
    &:focus,
    &:active {
      color: color(hig-blue-30);
      svg * {
        stroke: color(hig-blue-30);
      }
    }
  }
}
