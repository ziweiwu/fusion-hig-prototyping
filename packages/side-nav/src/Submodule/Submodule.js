import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import { sizes as iconSizes } from "@hig/icon";
import { ThemeContext } from "@hig/themes";
import ExternalLinkIcon from "../presenters/ExternalLinkIcon";

import "./submodule.scss";

export default class Submodule extends Component {
  static propTypes = {
    /** Indicates this submodule is currently active */
    active: PropTypes.bool,
    /** URL to navigate to when clicking this submodule */
    link: PropTypes.string,
    /** Called when clicking on the submodule */
    onClick: PropTypes.func,
    /** Called when hovering over the submodule */
    onMouseOver: PropTypes.func,
    /** Anchor target. Applicable only if link is provided */
    target: PropTypes.oneOf(["_self", "_blank", "_parent", "_top"]),
    /** Text to render */
    title: PropTypes.string
  };

  _renderExternalLinkIcon = () =>
    this.props.target === "_blank" && (
      <ExternalLinkIcon size={iconSizes.PX_16} />
    );

  render() {
    const { active, title, link, onClick, onMouseOver, target } = this.props;

    const classes = themeClass =>
      cx(themeClass, "hig__side-nav__submodule", {
        "hig__side-nav__submodule--active": active
      });

    const Wrapper = link ? "a" : "div";

    return (
      <ThemeContext.Consumer>
        {({ themeClass }) => (
          <Wrapper
            className={classes(themeClass)}
            href={link}
            target={target}
            onClick={onClick}
            onMouseOver={onMouseOver}
          >
            {title}
            {this._renderExternalLinkIcon()}
          </Wrapper>
        )}
      </ThemeContext.Consumer>
    );
  }
}
