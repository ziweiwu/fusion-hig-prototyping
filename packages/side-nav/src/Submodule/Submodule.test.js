import React from "react";
import renderer from "react-test-renderer";

import Submodule from "./Submodule";

describe("side-nav/Submodule", () => {
  describe("snapshot tests", () => {
    const cases = [
      {
        description: "renders with no props",
        props: {}
      }
    ];

    cases.forEach(({ description, props: { children, ...otherProps } }) => {
      it(description, () => {
        const wrapper = <Submodule {...otherProps}>{children}</Submodule>;
        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
