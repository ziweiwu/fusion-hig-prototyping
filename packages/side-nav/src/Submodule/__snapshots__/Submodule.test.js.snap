// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side-nav/Submodule snapshot tests renders with no props 1`] = `
<div
  className="hig--light-theme hig__side-nav__submodule"
  href={undefined}
  onClick={undefined}
  onMouseOver={undefined}
  target={undefined}
/>
`;
